vnext/Microsoft.ReactNative.IntegrationTests/JsiTurboModuleTests.cpp

```diff
@@ -0,0 +1,419 @@
+      rt, [error](facebook::jsi::Runtime &rt2, std::shared_ptr<facebook::react::Promise> promise) {
+        if (error) {
+          promise->reject("intentional promise rejection");
+        } else {
+          promise->resolve(facebook::jsi::String::createFromUtf8(rt2, "result!"));
+        }
+      });
+}
+
+facebook::jsi::Object MySimpleTurboModule::getConstants(facebook::jsi::Runtime &rt) {
+  LogAction("getConstants called", "OK");
+  // Note: return type isn't type-safe.
+  facebook::jsi::Object result(rt);
+  result.setProperty(rt, "const1", facebook::jsi::Value(true));
+  result.setProperty(rt, "const2", facebook::jsi::Value(375));
+  result.setProperty(rt, "const3", facebook::jsi::String::createFromUtf8(rt, "something"));
+  return result;
+}
+
+struct MySimpleTurboModulePackageProvider
+    : winrt::implements<MySimpleTurboModulePackageProvider, IReactPackageProvider> {
+  void CreatePackage(IReactPackageBuilder const &packageBuilder) noexcept {
+    AddTurboModuleProvider<MySimpleTurboModule>(packageBuilder, L"MySimpleTurboModule");
+  }
+};
+
+} // namespace
+
+TEST_CLASS (JsiTurboModuleTests) {
+  TEST_METHOD(ExecuteSampleTurboModule) {
+    MySimpleTurboModule::s_actionIndex = -1;
+    TestAction expectedActions[] = {
+        TestAction{"voidFunc called", nullptr},
+        TestAction{"getBool called", true},
+        TestAction{"getBool result", true},
+        TestAction{"getBool called", false},
+        TestAction{"getBool result", false},
+        TestAction{"getNumber called", 5.0},
+        TestAction{"getNumber result", 5.0},
+        TestAction{"getNumber called", NAN},
+        TestAction{"getNumber result", NAN},
+        TestAction{"getNumber called", INFINITY},
+        TestAction{"getNumber result", INFINITY},
+        TestAction{"getString called", "Hello"},
+        TestAction{"getString result", "Hello"},
+        TestAction{"getString called", ""},
+        TestAction{"getString result", ""},
+        TestAction{"getArray called", 3},
+        TestAction{"getArray result", "OK"},
+        TestAction{"getObject called", "OK"},
+        TestAction{"getObject result", "OK"},
+        TestAction{"getValue called", "OK"},
+        TestAction{"getValue result", "OK"},
+        TestAction{"getConstants called", "OK"},
+        TestAction{"getConstants result", "OK"},
+        TestAction{"getValueWithCallback called", "OK"},
+        TestAction{"getValueWithCallback result", "value from callback!"},
+        TestAction{"getValueWithPromise called", false},
+        TestAction{"getValueWithPromise called", true},
+        TestAction{"getValueWithPromise result resolve", "result!"},
+        TestAction{"getValueWithPromise result reject", "intentional promise rejection"},
+    };
+
+    ReactNativeHost host{};
+
+    auto queueController = winrt::Windows::System::DispatcherQueueController::CreateOnDedicatedThread();
+    queueController.DispatcherQueue().TryEnqueue([&]() noexcept {
```

Comment:
can you make sure this continues to work for winui 3? (use `Microsoft::System` instead of `Windows::System` if `USE_WINUI3` is defined)

Reply:
I was not aware that the `DispatcherQueue` in WinUI 3 uses a different namespace. I expected it to be more fundamental and not to depend on UI. Supporting it will require a quite extensive change and it must be subject of a different PR. I have created Issue #6805 to address it.